R/pkg/R/SQLContext.R

```diff
@@ -147,6 +147,30 @@ getDefaultSqlSource <- function() {
   l[["spark.sql.sources.default"]]
 }
 
+writeToTempFileInArrow <- function(rdf, numPartitions) {
+  stopifnot(require("arrow", quietly = TRUE))
+  stopifnot(require("withr", quietly = TRUE))
+  numPartitions <- if (!is.null(numPartitions)) {
+    numToInt(numPartitions)
+  } else {
+    1
+  }
+  fileName <- tempfile()
```

Comment:
might need to give it a dir prefix to use - the tempfile default is not CRAN compliant and possibly some ACL issue